src/Razor/Microsoft.AspNetCore.Razor.Language/src/RazorCodeDocumentExtensions.cs

```diff
@@ -250,14 +250,12 @@ public static bool TryComputeNamespace(this RazorCodeDocument document, bool fal
             }
 
             var filePath = document.Source.FilePath;
-            var relativePath = document.Source.RelativePath;
-            if (filePath == null || relativePath == null || filePath.Length < relativePath.Length)
+            if (filePath == null || document.Source.RelativePath == null || filePath.Length < document.Source.RelativePath.Length)
             {
                 @namespace = null;
                 return false;
             }
 
-            relativePath = NormalizePath(relativePath);
```

Comment:
`NormalizePath` turned `\\` to `/`. But turns out we were later splitting on either path separators so we can safely remove it.